driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -993,14 +994,15 @@ protected void channelRead0(ChannelHandlerContext ctx, Message.Response response
             ResponseHandler handler = pending.remove(streamId);
             streamIdHandler.release(streamId);
             if (handler == null) {
-                /**
+                /*
                  * During normal operation, we should not receive responses for which we don't have a handler. There is
                  * two cases however where this can happen:
                  *   1) The connection has been defuncted due to some internal error and we've raced between removing the
                  *      handler and actually closing the connection; since the original error has been logged, we're fine
                  *      ignoring this completely.
                  *   2) This request has timed out. In that case, we've already switched to another host (or errored out
                  *      to the user). So log it for debugging purpose, but it's fine ignoring otherwise.
+                 *   3) The connection encountered a frame too large to process.
```

Comment:
You are right, I was more commenting on the fact that this is a scenario where there is no handler, but it does seem wrong for this to be here since when this happens it'd be handled in `exceptionCaught()` like you said.   I'll fix this 👍